src/lib.rs

```diff
@@ -71,7 +71,8 @@ pub fn run(input: &str) -> LangResult<lir::Term> {
 pub fn run_with_machine<W: Write>(input: &str, mut machine: Machine<W>) -> LangResult<lir::Term> {
     let ast = parser::parse(input)?;
     let mir = mir::Term::from_ast(ast)?;
-    let _ty = ty::ty_check(&mir)?;
+    let ty = ty::ty_check(&mir)?;
+    println!("{}", ty);
```

Comment:
Don't forget to remove this print debugging.
PS: For cases like this the `dbg!`[[1](https://doc.rust-lang.org/std/macro.dbg.html)] macro is usually easier to use and easier to find later.

Reply:
I force pushed my last commit fixing this.... I'm embarrased :P

PS: I think we could implement some sort of debug mode into the compiler to avoid this in the future.